pkg/core/index.go

```diff
@@ -250,8 +248,8 @@ func (index *Index) UpdateMetadataByShard(n int64) {
 	index.lock.Unlock()
 }
 
-func (index *Index) UpdateStatsBySecondShard(n, secondIndex int64) {
-	shard := index.shards[n]
+func (index *Index) UpdateStatsBySecondShard(id string, secondIndex int64) {
```

Comment:
Can you add comments to every function saying what it is supposed to do.

Reply:
Okay